sqlmesh/core/engine_adapter/base.py

```diff
@@ -59,6 +60,7 @@ class EngineAdapter:
     DEFAULT_BATCH_SIZE = 10000
     DEFAULT_SQL_GEN_KWARGS: t.Dict[str, str | bool | int] = {}
     ESCAPE_JSON = False
+    SCHEMA_DIFF_CONFIG = SchemaDiffConfig()
```

Comment:
Should this be `TABLE_ALTERATION_CONFIG`? Seems like we should have 2 configs: 1 which determines how to compute the diff and another - how it should be applied. Currently it feels like there is a tight coupling between diffing and engine adapter.

Reply:
Ok, seems like this config is not actually required when computing the diff itself. Should this then be `TableAlterationConfig`?

Reply:
So technically right now the diff calculation is universal and is not specific to any engine. In other words the diff can be calculated without any context of what engine is applying it. These configs are then defined in the engine adapter since they only are related to how the diff is applied. If requirements change or use cases to where we need the config when calculating the diff then we can pass the config into the diff itself and we will start making engine specific diffs but then application would not require addition config.